src/server/controller/cycleData/links/visitCycleLinks/visitCycleLinks.ts

```diff
@@ -11,12 +11,15 @@ const jobOptions: JobsOptions = {
   removeOnFail: true,
 }
 
-export const visitCycleLinks = (props: VisitCycleLinksProps): Promise<Job<VisitCycleLinksProps>> | undefined => {
+export const visitCycleLinks = async (props: VisitCycleLinksProps): Promise<Job<VisitCycleLinksProps>> | undefined => {
   const { assessment, cycle } = props
 
   Logger.debug(`[visitCycleLinks] added visit all links job for ${assessment.props.name} ${cycle.name} to the queue`)
 
-  // TODO: persist when this function is already running, and if so, return early.
+  const activeJobs = await VisitCycleLinksQueueFactory.getActiveJobs(props)
+  const isVerificationInProgress = activeJobs.length > 0
+
+  if (isVerificationInProgress) return undefined
```

Comment:
We already disable the button in the FE in case the job is already running, but adding the check here as well in case of potential double clicks.